src/ray/ray_config_def.h

```diff
@@ -0,0 +1,132 @@
+// Note: below are the definitions of all config items. 
+// The Macro DEFINE_CONFIG is not defined in this fine.
+// The user can define there only Macro to use the information
+// and avoid duplicated code.
```

Comment:
This might explain more clearly. 

> This header file is used to avoid code duplication, it could be included multiple times in `ray_config.h`, 
> with each place having a different definition of the `DEFINE_CONFIG` macro. 

Also, maybe rename `DEFINE_CONFIG` to `RAY_CONFIG`, because the macro doesn't always define a config now.